### DISCUSSION TITLE
slow cloning

### DISCUSSION THREAD

**Discussion by advancedwebdeveloper** [community/community#33699] [Other Feature Feedback, Questions, & Ideas]
Hello.

I have slow cloning rates

> $ git clone https://github.com/kubernetes/minikube
> Cloning into 'minikube'...
> remote: Enumerating objects: 185636, done.
> remote: Counting objects: 100% (964/964), done.
> remote: Compressing objects: 100% (693/693), done.
> Receiving objects:  12% (23819/185636), 75.46 MiB | 83.00 KiB/s 

> $ git clone https://github.com/github/debug-repo /tmp/debug-repo-http
> git clone git@github.com:github/debug-repo /tmp/debug-repo-ssh
> ping -c 10 github.com
> traceroute github.com
> Cloning into '/tmp/debug-repo-http'...
> remote: Enumerating objects: 1653, done.
> remote: Counting objects: 100% (4/4), done.
> remote: Compressing objects: 100% (4/4), done.
> remote: Total 1653 (delta 0), reused 0 (delta 0), pack-reused 1649
> Receiving objects: 100% (1653/1653), 329.28 KiB | 113.00 KiB/s, done.
> Resolving deltas: 100% (1063/1063), done.
> Cloning into '/tmp/debug-repo-ssh'...
> The authenticity of host 'github.com (140.82.121.3)' can't be established.
> ED25519 key fingerprint is SHA256:+DiY3wvvV6TuJJhbpZisF/zLDA0zPMSvHdkr4UvCOqU.
> This key is not known by any other names
> Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
> Warning: Permanently added 'github.com' (ED25519) to the list of known hosts.
> remote: Enumerating objects: 1653, done.
> remote: Counting objects: 100% (4/4), done.
> remote: Compressing objects: 100% (4/4), done.
> remote: Total 1653 (delta 0), reused 0 (delta 0), pack-reused 1649
> Receiving objects: 100% (1653/1653), 329.28 KiB | 102.00 KiB/s, done.
> Resolving deltas: 100% (1063/1063), done.
> PING github.com (140.82.121.3) 56(84) bytes of data.
> 64 bytes from lb-140-82-121-3-fra.github.com (140.82.121.3): icmp_seq=1 ttl=53 time=50.3 ms
> 64 bytes from lb-140-82-121-3-fra.github.com (140.82.121.3): icmp_seq=2 ttl=53 time=48.0 ms
> 64 bytes from lb-140-82-121-3-fra.github.com (140.82.121.3): icmp_seq=3 ttl=53 time=46.1 ms
> 64 bytes from lb-140-82-121-3-fra.github.com (140.82.121.3): icmp_seq=4 ttl=53 time=45.4 ms
> 64 bytes from lb-140-82-121-3-fra.github.com (140.82.121.3): icmp_seq=5 ttl=53 time=43.8 ms
> 64 bytes from lb-140-82-121-3-fra.github.com (140.82.121.3): icmp_seq=6 ttl=53 time=49.9 ms
> 64 bytes from lb-140-82-121-3-fra.github.com (140.82.121.3): icmp_seq=7 ttl=53 time=47.9 ms
> 64 bytes from lb-140-82-121-3-fra.github.com (140.82.121.3): icmp_seq=8 ttl=53 time=46.4 ms
> 64 bytes from lb-140-82-121-3-fra.github.com (140.82.121.3): icmp_seq=9 ttl=53 time=48.9 ms
> 64 bytes from lb-140-82-121-3-fra.github.com (140.82.121.3): icmp_seq=10 ttl=53 time=53.3 ms
> 
> --- github.com ping statistics ---
> 10 packets transmitted, 10 received, 0% packet loss, time 9014ms
> rtt min/avg/max/mdev = 43.789/48.003/53.338/2.635 ms
> traceroute to github.com (140.82.121.3), 30 hops max, 60 byte packets
>  1  _gateway (192.168.12.62)  0.614 ms  0.941 ms  1.157 ms
>  2  10.74.37.41 (10.74.37.41)  30.494 ms * *
>  3  10.74.37.33 (10.74.37.33)  30.398 ms  30.362 ms  30.450 ms
>  4  * * *
>  5  77.75.146.147 (77.75.146.147)  34.706 ms  34.674 ms  34.643 ms
>  6  77.75.146.146 (77.75.146.146)  34.476 ms  26.988 ms  25.291 ms
>  7  * * *
>  8  * * *
>  9  de-cix.fra.github.com (80.81.196.79)  53.438 ms  56.310 ms  56.289 ms
> 10  * * *
> 11  * * *
> 12  * * *
> 13  * * *
> 14  * * *
> 15  * * *
> 16  * * *
> 17  * * *
> 18  * * *
> 19  * * *
> 20  * * *
> 21  * * *
> 22  * * *
> 23  * * *
> 24  * * *
> 25  * * *
> 26  * * *
> 27  * * *
> 28  * * *
> 29  * * *
> 30  * * *
>

**Comment by Matteo-it** [comment]
When i had to work in hotspot i had the same problem. Contact the https://support.github.com/request. 

Another option (both is good!) is to contact your ISP and ask why their route to GitHub is slow.
